File: 01-OOP-Refresher/src/go/circle.go
package main

import (
	"fmt"
	"math"
)

type Circle struct {
	Radius float64
}

func (c *Circle) Draw() {
	fmt.Printf("Drawing Circle with radius %.1f\n", c.Radius)
}

func (c *Circle) Area() float64 {
	return math.Pi * c.Radius * c.Radius
}

File: 01-OOP-Refresher/src/rust/circle.rs
use std::f64::consts::PI;
use crate::drawable::Drawable;

pub struct Circle {
    pub radius: f64,
}

impl Drawable for Circle {
    fn draw(&self) {
        println!("Drawing Circle with radius {}", self.radius);
    }

    fn area(&self) -> f64 {
        PI * self.radius * self.radius
    }
}
